cmd/pinniped-proxy/src/service.rs

```diff
@@ -9,11 +10,36 @@ use crate::https;
 pub async fn proxy(req: Request<Body>) -> Result<Response<Body>, Infallible> {
     let log_data = logging::request_log_data(&req);
 
+    let _ = match https::get_api_server_url(req.headers()) {
+        Ok(u) => u,
+        Err(e) => return handle_error(e, log_data),
+    };
+    // TODO: don't call this if we're using https://kubernetes.local, instead
+    // grab the data from the file system.
+    let cert_auth_data = match https::get_api_server_cert_auth_data(req.headers()) {
```

Comment:
who was setting this auth data? where is the cert coming from?

Reply:
In our specific case, the Kubeapps frontend, which has the multi-cluster config (collection of api server urls together with the cert auth data for each, where required), adds these headers when forwarding the request to pinniped-proxy. This allows pinniped-proxy to receive the request, exchange the credential, then forward the request on to the api server (with the cert auth) together with the client auth certs.

Reply:
I see, but this is not implemented yet, no? (or I am missing something?)

Reply:
Correct - Currently the service is just logging requests - I'll add the proxying and credential exchange next.